### PR TITLE
Update documentation about state ordering. (Fixes #45288)

### LINKED ISSUE #45288: Docs about Ordering States
### Description of Issue/Question
I think the current docs of "Ordering states" could be improved.

> Salt always executes states in a finite manner, meaning that they will always execute in the same order regardless of the system that is executing them. But in Salt 0.17.0, the state_auto_order option was added. This option makes states get evaluated in the order in which they are defined in sls files, including the top.sls file.

>The evaluation order makes it easy to know what order the states will be executed in, but it is important to note that the requisite system will override the ordering defined in the files, and the order option described below will also override the order in which states are defined in sls files.

> If the classic ordering is preferred (lexicographic), then set state_auto_order to False in the master configuration file. Otherwise, state_auto_order defaults to True.

Source https://docs.saltstack.com/en/latest/ref/states/ordering.html#state-auto-ordering

The version 0.17.0 was released on 2013-09-26. I guess 99% of all users use a newer version.

I think it would be much more easier to read, if you just present the current state exclusively. Users of the four years old version should read the four years old docs :-)

This is current state: States get evaluated in the order in which they are defined in sls files.

### PR DESCRIPTION
**Pull request by MTecknology**
### What does this PR do?

This is a simple documentation update.

### What issues does this PR fix or reference?

- #45288

### Commits signed with GPG?

Yes

### CODE REVIEW

File: doc/ref/states/ordering.rst [hunk 1]
```diff
@@ -24,23 +24,18 @@ dependencies via the `requisite` system.
 State Auto Ordering
 ===================
 
-.. versionadded: 0.17.0
```
Comment (rallytime) [inline doc/ref/states/ordering.rst]: Why are you removing this?

File: doc/ref/states/ordering.rst [hunk 2]
```diff
@@ -24,23 +24,18 @@ dependencies via the `requisite` system.
...
-But in Salt 0.17.0, the ``state_auto_order`` option was added. This option
-makes states get evaluated in the order in which they are defined in sls
-files, including the top.sls file.
-
-The evaluation order makes it easy to know what order the states will be
-executed in, but it is important to note that the requisite system will
-override the ordering defined in the files, and the ``order`` option described
-below will also override the order in which states are defined in sls files.
+execute in the same order regardless of the system that is executing them. This
+evaluation order makes it easy to know what order the states will beexecuted in,
```
Comment (rallytime) [inline doc/ref/states/ordering.rst]: Typo `beexecuted` --> `be executed`

**Review by rallytime — COMMENTED** [review]
Some small comments :)

File: doc/ref/states/ordering.rst (hunk 2 above)
Comment (MTecknology) [reply doc/ref/states/ordering.rst]: Thanks for catching that!

File: doc/ref/states/ordering.rst (hunk 1 above)
Comment (MTecknology) [reply doc/ref/states/ordering.rst]: According to (modified) version history, I would never ever have done such a silly thing! ;)

**Review by rallytime — APPROVED** [review]

**Pull request merged**